README.md

```diff
@@ -7,11 +7,60 @@
 
 ## Overview
 
-This lesson is a very high-level over view of the React component lifecycle. We
-want to walk students through the whole shebang.
+As you've learned previously, a React component has two sets of properties: **props** and **state**. Props are given to the component by it's parent. You can think of props as an external influence that the component has no control over - like the weather. The other set of properties the component has is called the state. A component's state is internal to the component and it can change when the props changing or the user interacting with the component. For example, if the weather went from sunny to raining, you would put on a raincoat and wellies. In React component terms, your state could go from "wearing a jacket" to "wearing a raincoat and wellies". You know to put on a raincoat when it's raining because you've been told by your parents. A React component would have to be told what to do by the programmer - that's you! :)
```

Comment:
> As you've learned previously

We try to decouple lessons as much as possible, so in general, it's best to not refer to any 'previous' lessons — it makes sense in this course, but lessons might be pulled out and reordered.

> and it can change when the props changing
> ... For example, if the weather went from sunny to raining, you would put on a raincoat and wellies.
> In React component terms, your state could go from "wearing a jacket" to "wearing a raincoat and wellies".

State changes on prop changes should never really happen, we use computed values for those instead of adding it to the state — can we remove this? Might be a little confusing.

TIL what wellies are! 👢